### ISSUE TITLE
Terminal went full transparent after repeated Show/Hide

### ISSUE THREAD

**Issue by h-collector** [linvinus/AltYo#25]
After latest update via ppa, that is to version altyo_0.4~rc16-linvinus1_amd64 I'm observing really strange issue:
After repeated hiding and showing terminal, it's getting fully transparent. 

I can get Context Popup Menu via RMB (execute show, quit) if height is  set to 100%, but that's all, it stay fully transparent: no tab bar and no terminal visible.

Reverting settings to factory defaults don't help, also with plugged second monitor and after restart, issue remains.
No errors reported cinnamon log or via dmesg.

My system specs are:
Linux Mint 17.1 Cinnamon 64bit, with latest packages (dist-upgrade) and Oibaff Graphics Drivers.

The strange thing is that altyo_0.4~rc5 (previous available version on ppa) works just fine.

If it helps, debug output (show/hide events):
```
domain:(null) message:main.vala:133: workaround != w_file 0 != 0
domain:(null) message:main.vala:527: git_hash=
domain:(null) message:main.vala:528: changelog_tag=0.4_rc16-linvinus1
domain:(null) message:altyo_window.vala:850: reconfigure VTWindow
domain:(null) message:altyo_hotkey.vala:134: bind Menu display=24901440 key_sym=65383 modifiers=0
domain:(null) message:altyo_window.vala:1352: AYObject new
domain:(null) message:altyo_window.vala:1502: reconfigure AYObject
[..]
domain:(null) message:altyo_window.vala:971: x=0,y=0
domain:(null) message:altyo_window.vala:997: monitor name LVDS 0
domain:(null) message:altyo_window.vala:1004: monitor_workarea x=0,y=25 w=1366 h=743
domain:(null) message:altyo_window.vala:1027: settings for monitor x=1,y=-1 w=100 h=50
domain:(null) message:altyo_window.vala:1080: configure_position end x=0,y=25 term_w=1366 term_h=371
domain:(null) message:altyo_window.vala:1204: update_position_size start maximized=0 this.my_window_state=0
domain:(null) message:altyo_window.vala:2770: maximize==START
domain:(null) message:altyo_window.vala:1219: terminal w=1366 h=371
domain:(null) message:altyo_window.vala:2797: get_altyo_height terminal_width=1366 should_be_h=373
domain:(null) message:altyo_window.vala:1236: update_position_size should_be_h=373 terminal_width=1366 x=0 y=25
domain:(null) message:altyo_window.vala:332: CreateVTWindow end
domain:(null) message:hvbox.vala:383: size_allocate x=0 y=371 w=1366 h=2

domain:(null) message:altyo_window.vala:1204: update_position_size start maximized=0 this.my_window_state=0
domain:(null) message:altyo_window.vala:1219: terminal w=1366 h=371
domain:(null) message:altyo_window.vala:2797: get_altyo_height terminal_width=1366 should_be_h=373
domain:(null) message:altyo_window.vala:1236: update_position_size should_be_h=373 terminal_width=1366 x=0 y=25
domain:(null) message:altyo_window.vala:368: end win show
domain:(null) message:altyo_window.vala:691: window_state_event !!!!!!!!! this.maximized=0 event.changed_mask=129
domain:(null) message:altyo_window.vala:691: window_state_event !!!!!!!!! this.maximized=0 event.changed_mask=128
domain:(null) message:hvbox.vala:383: size_allocate x=0 y=371 w=1366 h=2

domain:(null) message:altyo_terminal.vala:203: set_title start
domain:(null) message:altyo_terminal.vala:207: set_title(0,(null))
domain:(null) message:altyo_terminal.vala:1054: url_regexps=2
domain:(null) message:altyo_terminal.vala:1057: match_add_gregex 0
domain:(null) message:altyo_terminal.vala:571: AYTerm: palette-0:rgb(0,0,0)
domain:(null) message:altyo_terminal.vala:571: AYTerm: palette-1:rgb(170,0,0)
domain:(null) message:altyo_terminal.vala:571: AYTerm: palette-2:rgb(0,170,0)
domain:(null) message:altyo_terminal.vala:571: AYTerm: palette-3:rgb(170,84,0)
domain:(null) message:altyo_terminal.vala:571: AYTerm: palette-4:rgb(0,0,170)
domain:(null) message:altyo_terminal.vala:571: AYTerm: palette-5:rgb(170,0,170)
domain:(null) message:altyo_terminal.vala:571: AYTerm: palette-6:rgb(0,170,170)
domain:(null) message:altyo_terminal.vala:571: AYTerm: palette-7:rgb(170,170,170)
domain:(null) message:altyo_terminal.vala:571: AYTerm: palette-8:rgb(84,84,84)
domain:(null) message:altyo_terminal.vala:571: AYTerm: palette-9:rgb(255,84,84)
domain:(null) message:altyo_terminal.vala:571: AYTerm: palette-10:rgb(84,255,84)
domain:(null) message:altyo_terminal.vala:571: AYTerm: palette-11:rgb(255,255,84)
domain:(null) message:altyo_terminal.vala:571: AYTerm: palette-12:rgb(84,84,255)
domain:(null) message:altyo_terminal.vala:571: AYTerm: palette-13:rgb(255,84,255)
domain:(null) message:altyo_terminal.vala:571: AYTerm: palette-14:rgb(84,255,255)
domain:(null) message:altyo_terminal.vala:571: AYTerm: palette-15:rgb(255,255,255)
domain:(null) message:altyo_terminal.vala:571: AYTerm: fg-color:rgb(170,170,170)
domain:(null) message:altyo_terminal.vala:571: AYTerm: bg-color:rgba(0,0,0,0.80000000000000004)
domain:(null) message:altyo_terminal.vala:762: run command:/bin/bash
domain:(null) message:altyo_terminal.vala:807: exclude:WINDOWID
domain:(null) message:altyo_terminal.vala:807: exclude:COLORTERM
domain:(null) message:hvbox.vala:550: inser at 1

domain:(null) message:altyo_window.vala:793: check_focusout focus=1 state=0
domain:(null) message:altyo_window.vala:793: check_focusout focus=1 state=0
domain:(null) message:altyo_window.vala:691: window_state_event !!!!!!!!! this.maximized=0 event.changed_mask=128
domain:(null) message:altyo_window.vala:793: check_focusout focus=1 state=0
domain:(null) message:hvbox.vala:383: size_allocate x=0 y=371 w=1366 h=21

domain:(null) message:hvbox.vala:719: draw

domain:(null) message:altyo_window.vala:662: toggle_window start
domain:(null) message:altyo_window.vala:672: slf=52428804 w=52428804
domain:(null) message:altyo_window.vala:568: pull_up orig_h=392 orig_w=1366
domain:(null) message:altyo_window.vala:573: pull_up orig_h_tasks_notebook=371 orig_w_tasks_notebook=1366
domain:(null) message:hvbox.vala:383: size_allocate x=0 y=371 w=1366 h=21

domain:(null) message:hvbox.vala:719: draw

domain:(null) message:altyo_window.vala:241: pixwin.damage_event
domain:(null) message:altyo_window.vala:684: toggle_window end
domain:(null) message:altyo_window.vala:241: pixwin.damage_event
domain:(null) message:altyo_window.vala:691: window_state_event !!!!!!!!! this.maximized=0 event.changed_mask=1
domain:(null) message:altyo_window.vala:691: window_state_event !!!!!!!!! this.maximized=0 event.changed_mask=128
domain:(null) message:altyo_window.vala:793: check_focusout focus=0 state=1
domain:(null) message:altyo_window.vala:793: check_focusout focus=0 state=1
domain:(null) message:altyo_window.vala:793: check_focusout focus=0 state=1
domain:(null) message:altyo_window.vala:662: toggle_window start
domain:(null) message:altyo_window.vala:672: slf=52428804 w=67109465
domain:(null) message:altyo_window.vala:971: x=0,y=25
domain:(null) message:altyo_window.vala:997: monitor name LVDS 0
domain:(null) message:altyo_window.vala:1004: monitor_workarea x=0,y=25 w=1366 h=743
domain:(null) message:altyo_window.vala:1027: settings for monitor x=1,y=-1 w=100 h=50
domain:(null) message:altyo_window.vala:1080: configure_position end x=0,y=25 term_w=1366 term_h=371
domain:(null) message:altyo_window.vala:684: toggle_window end
domain:(null) message:altyo_window.vala:691: window_state_event !!!!!!!!! this.maximized=0 event.changed_mask=129
domain:(null) message:altyo_window.vala:691: window_state_event !!!!!!!!! this.maximized=0 event.changed_mask=128
domain:(null) message:altyo_window.vala:793: check_focusout focus=1 state=0
domain:(null) message:altyo_window.vala:793: check_focusout focus=1 state=0
domain:(null) message:altyo_window.vala:691: window_state_event !!!!!!!!! this.maximized=0 event.changed_mask=128
domain:(null) message:altyo_window.vala:793: check_focusout focus=1 state=0
domain:(null) message:altyo_window.vala:450: on_pull_down last step 0
domain:(null) message:altyo_window.vala:470: on_pull_down very last step
domain:(null) message:altyo_window.vala:1204: update_position_size start maximized=0 this.my_window_state=0
domain:(null) message:altyo_window.vala:1219: terminal w=1366 h=371
domain:(null) message:altyo_window.vala:2797: get_altyo_height terminal_width=1366 should_be_h=392
domain:(null) message:altyo_window.vala:1236: update_position_size should_be_h=392 terminal_width=1366 x=0 y=25
domain:(null) message:altyo_window.vala:793: check_focusout focus=1 state=0
domain:(null) message:altyo_window.vala:691: window_state_event !!!!!!!!! this.maximized=0 event.changed_mask=128
domain:(null) message:altyo_window.vala:793: check_focusout focus=0 state=0
domain:(null) message:altyo_window.vala:793: check_focusout focus=0 state=0
domain:(null) message:altyo_window.vala:793: check_focusout focus=0 state=0
```

**Comment by LaraCraft93** [comment]
I have the same problem! (Gentoo x86_64, xfce4). After some time of use, The altyo become completely invisible. If I go to fullscreen (F11) and right click on invisible area, I can view the context menu, but I cannot see the terminal, only if I do "killall altyo" and start again.

I found the following output on dmesg:

[ 9518.239990] altyo[4709]: segfault at 73a7678 ip 0000000000421728 sp 00007fffc47dde70 error 4 in altyo (deleted)[400000+c8000]

**Comment by LaraCraft93** [comment]
I found that if I increase the buffer size to 10000 or something similar, it takes longer for the problem happens

**Comment by linvinus** [comment]
>I found that if I increase the buffer size to 10000 or something similar, 

wow, very interesting, i will check that

**Comment by linvinus** [comment]
>I found that if I increase the buffer size to 10000 or something similar, 

i can't reproduce, but i saw that problem previously
i have no idea how to reproduce it, and why  it happens

**Comment by LaraCraft93** [comment]
I'll try to find a way to reproduce

**Comment by linvinus** [comment]
hi, i got it, but i don't know how to resolve.
main problem - draw event is absent (for some reason), all other events (mouse, keyboard) working  fine.
so, problem in GTK or in X server.
i have no idea how to resolve that :(
in my opinion, problem occur when not enough memory (system memory in heavy swapping)

**Comment by LaraCraft93** [comment]
in my opinion, problem occur when not enough memory (system memory in heavy swapping) 

I monitored that when the problem occurred, and can be discarded. I have 12gb of memory, and it was not being used at the time of failure.

**Comment by linvinus** [comment]
ok, thanks for report.
so what we have:
1) for some reason,after some action,  draw event is  skipped  (expose event in terms of X window)
2) Gtk window object and X window object exist and the same as previously and working fine
3) there is no any warning or error when problem happens (in debug log)
4) i can't force draw event from application (nothing happed)

time to time on my system,when i opening a new program, i have error "Maximum number of clients reached" from X server. (i really have a lot of opened windows in that case)
may be in that case something wrong happens in altyo too? 
do you have such error when problem occur?

**Comment by noirbizarre** [comment]
I have the same issue that make AltYo unreliable for daily usage.
Is there any news on fixing this one ?

**Comment by linvinus** [comment]
sorry, something go wrong inside GTK, i can't catch what is it.

**Comment by thankjura** [comment]
I have the same problem with last version from git! (Gentoo x86_64, gnome-shell)

**Comment by ShyPixie** [comment]
@linvinus I found a way to reproduce this problem. 

1. Clone the git repository, and compile with ``` make ```
2. execute ``` ./altyo ```
3. on bash shell, execute: ``` a=b; while true; do a+=b; echo $a; sleep .000001; done ```
4. while the script is running press alt+´ (show/hide key) several times

I think this is also the same reason for my terminal hangs after some compilations. And as I said earlier, increasing the buffer size makes the terminal takes longer to stop, then surely these things are related.

I hope this helps.

**Comment by linvinus** [comment]
thankjura, confirm.

ShyPixie , thank you, you are right, but i don't know how to find solution.

lets see, when there is no problem we can get back trace for "draw" event, remember it.
but when problem occur  there is no draw event.
In that case i use break points inside gdk but i  don't understand where the problem is.
if you know gdb enough you may try to find differences in behavior.  

i really spend a lot of time on that, with no luck

**Comment by linvinus** [comment]
how to debug
```
$ export VALA_FLAGS =" -g --save-temps -X -O0"
$ make
$ gdb ./altyo
break vt_main_window_real_draw
run
```

**Comment by FRiMN** [comment]
I have a similar (or the same) problem with Guake (0.4.4-1ubuntu1).

**Comment by linvinus** [comment]
Thank you for the information!

**Comment by linvinus** [comment]
after one month of testing i believe that this bug was fixed in commit 45d25ae

**Comment by h-collector** [comment]
Testing new build. It Works, thanks!
